R/utils_formatters.R

```diff
@@ -361,9 +358,174 @@ paste_currency_str <- function(x,
           direction = placement
         ) %>%
         paste_on_side(
-          x_side = currency_str,
+          x_side = symbol_str,
           direction = placement
         )
     }
   })
 }
+
+#' @noRd
+format_in_accounting_style <- function(x,
+                                       x_vals,
+                                       accounting,
+                                       minus_mark,
+                                       parens_marks) {
+
+  # Return values unchanged if there are no negative values
+  if (!any(x_vals < 0)) {
+    return(x)
+  }
+
+  # Handle replacement of the minus mark
+  x <-
+    x %>%
+    tidy_gsub("-", minus_mark, fixed = TRUE)
+
+  # Handle case where negative values are to be placed within parentheses
+  if (accounting) {
+
+    # Selectively remove minus sign and paste between parentheses
+    x[x_vals < 0] <-
+      x[x_vals < 0] %>%
+      tidy_gsub(minus_mark, "", fixed = TRUE) %>%
+      paste_between(x_2 = parens_marks)
+  }
+
+  x
+}
+
+#' @noRd
+prettify_scientific_notation <- function(x,
+                                         small_pos,
+                                         exp_marks,
+                                         minus_mark) {
+
+  if (!any(grepl("e|E", x))) {
+    return(x)
+  }
+
+  # For any numbers that shouldn't have an exponent, remove
+  # that portion from the character version
+  x[small_pos] <-
+    split_scientific_notn(x[small_pos])$num
+
+  # For any non-NA numbers that do have an exponent, format
+  # those according to the output context
+  sci_parts <- split_scientific_notn(x[!small_pos])
+
+  x[!small_pos] <-
+    paste0(
+      sci_parts$num, exp_marks[1],
+      sci_parts$exp, exp_marks[2]
+    )
+
+  # Handle replacement of the minus mark in number
+  # and exponent parts
+  x %>%
+    tidy_gsub("-", minus_mark, fixed = TRUE)
+}
+
+#' @noRd
+num_formatter_factory <- function(context,
+                                  decimals = NULL,
+                                  suffix_labels = NULL,
+                                  scale_by = NULL,
+                                  sep_mark = NULL,
+                                  dec_mark = NULL,
+                                  currency = NULL,
+                                  drop_trailing_zeros = NULL,
+                                  accounting = NULL,
+                                  incl_space = NULL,
+                                  placement = NULL,
+                                  pattern = NULL,
+                                  format_fn = NULL) {
+
+  force(context)
```

Comment:
Explore possibility of `force()`ing vars in a list.

Reply:
`force(list(a, b, c, ...))`

Reply:
@schloerke this seemed to work in my testing (no R error about syntax). However, do you know of any way to verify that the variables are indeed forced?

Reply:
I tested this to ensure that the forcing works with a list.